File: smart_lib/src/main/java/com/mitnick/tannotour/smart_lib/cache/observer/annos/CacheReceiver.java
package com.mitnick.tannotour.smart_lib.cache.observer.annos;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by mitnick on 2018/3/12.
 * Description
 */

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface CacheReceiver {
    Class cacheBean();
}

File: smart_lib/src/main/java/com/mitnick/tannotour/smart_lib/cache/observer/annos/CacheReceiver.kt
package com.mitnick.tannotour.smart_lib.cache.observer.annos

import com.mitnick.tannotour.smart_lib.cache.bean.CacheBean
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import kotlin.reflect.KClass

/**
 * Created by mitnick on 2018/3/12.
 * Description
 */

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(RetentionPolicy.RUNTIME)
annotation class CacheReceiver(val cacheBean: KClass<out CacheBean>)
